src/utils/ProfileProvider.tsx

```diff
@@ -0,0 +1,85 @@
+'use client';
+
+import supabase from '@/api/supabase/createClient';
+import { fetchProfileById } from '@/api/supabase/queries/profiles';
+import { Profile } from '@/types/schema';
+import { UUID } from 'crypto';
+import { createContext, useState, useEffect, useMemo, ReactNode } from 'react';
+
+interface ProfileContextType {
+  profileData: Profile | null;
+  updateProfile: (newProfileData: Profile) => void;
+}
+
+export const ProfileContext = createContext<ProfileContextType | undefined>(
+  undefined,
+);
+
+export default function ProfileProvider({ children }: { children: ReactNode }) {
+  const [profileData, setProfileData] = useState<Profile | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      const { data: sessionData, error } = await supabase.auth.getSession();
+
+      if (error) throw error;
+      if (
+        !sessionData ||
+        !sessionData.session ||
+        !sessionData.session.user ||
+        !sessionData.session.user.id
+      )
+        return;
+
+      const data = await fetchProfileById(sessionData.session.user.id as UUID);
+      setProfileData(data);
+    })();
+  }, []);
+
+  const providerValue = useMemo(
+    () => ({
+      profileData,
+      updateProfile: setProfileData,
```

Comment:
In the pseudocode on the Linear issue, the `updateProfile` function looks like this:

```
// Update the profile data locally and trigger an API call to update it on Supabase
  const updateProfile = async (newProfileData: Profile) => {
    // Update the local state
    setProfileData(newProfileData);

    // [pragya] call the supabase query you wrote to update the profile for this user
  };
```

Wondering if we might be missing the last part here— i.e., when `profileData` is updated, is the corresponding entry in the `profiles` table on Supabase also updated?

On the settings page, when an edit is made to the profile, I think we ideally want to be able to call the `updateProfile` in the `ProfileContext` and have both the `Profile` object in the context and the entry on Supabase be updated.

Reply:
oh yes thanks for the catch!